src/lib/src/select-container.component.ts

```diff
@@ -239,18 +246,45 @@ export class SelectContainerComponent implements AfterViewInit, OnDestroy {
   }
 
   private observeSelectableItems() {
-    // Update the container as well as all selectable items if the list has changed
-    this.$selectableItems.changes.pipe(takeUntil(this.destroy$)).subscribe((items: QueryList<SelectItemDirective>) => {
-      const newList = items.toArray();
-      const removedItems = this._selectedItems.filter(item => !newList.includes(item.value));
+    // Listen for update actions and either select or deselect an item
+    this.updateItems$
+      .pipe(
+        mergeMap(update => {
+          const item = update.item;
+
+          switch (update.type) {
```

Comment:
Could this be written "more reactive" by using the `partition` operator?
Not saying you should tho, just asking because I use this lib to step up my RxJS game 😅

Reply:
Yep, you can also use `partition` to split the updates into two buckets. But then you end up with two streams and you'd have two subscriptions. Subscriptions in general is a rather expensive operation. Sometimes it's better to just have a conditional and distinguish there instead of splitting your stream into more streams.

Reply:
Ah gotcha, thanks!